src/components/ToggleSwitch/ToggleSwitch.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import { IToggleSwitchProps } from './ToggleSwitch.Props';
+import './ToggleSwitch.scss';
+import { autobind } from '../../utilities/autobind';
+
+export class ToggleSwitch extends React.Component<IToggleSwitchProps, any> {
+    constructor(props) {
+        super(props);
+        this.state = {
+            checked : props.checked || false
+        };
+    }
+
+    public static defaultProps : IToggleSwitchProps = {
+        checked : false
+    };
+
+    public render() : JSX.Element {
+        let{
+            checked,
+            onChange,
+            className
+        } = this.props;
+
+        const switchClassName = classNames(
+            
+            className,
+            'toggle-switch'
+        );
+        const slidersClassName = classNames(
+            'toggle-slider'
+        );
+        return(
+            <label className={switchClassName}>
+                <input type="checkbox" onChange={this._onChange} checked={this.state.checked}/>
```

Comment:
you are using state here, but what if props.checked is not undefined? we should use the props value here in such a case. or even better, shouldn't we calculate the state on componentwillreceiveprops so that the state is in sync across the component? There is another component that does something like this. I believe it is the default checkbox component.